src/BayesFilters/include/BayesFilters/ParticleSet.h

```diff
@@ -0,0 +1,36 @@
+#ifndef PARTICLESET_H
+#define PARTICLESET_H
+
+#include <BayesFilters/GaussianMixture.h>
+
+#include <vector>
+
+namespace bfl {
+    class ParticleSet;
+}
+
+
+class bfl::ParticleSet : public bfl::GaussianMixture
+{
+public:
+    ParticleSet(const std::size_t components, const std::size_t dim) noexcept;
+
+    ParticleSet(const std::size_t components, const std::size_t dim_linear, const std::size_t dim_circular) noexcept;
+
+    virtual ~ParticleSet() noexcept;
+
+    ParticleSet operator+(const ParticleSet& rhs);
```

Comment:
I would suggest to also implement `operator+=` and to follow guidelines reported [here](https://en.cppreference.com/w/cpp/language/operators).